types/lodash/common/object.d.ts

```diff
@@ -1327,6 +1327,7 @@ declare module "../index" {
          * _.has(other, 'a');
          * // => false
          */
+        has<T, K extends PropertyName>(object: T, path: K): object is T & Record<string, unknown> & { [P in K]: P extends keyof T ? T[P] : Record<string, unknown> extends T ? T[keyof T] : unknown};
```

Comment:
This return type is getting quite complex so let's see if we can simplify it:

Can the first `Record<string, unknown> &` be removed? I don't think it does anything valuable.

You might also be able to remove the `T &` from the beginning since I think it's automatically implied by the type guard. Not sure though - if it makes your tests fail then feel free to ignore this.